Common/include/omp_structure.hpp

```diff
@@ -122,3 +126,49 @@ inline size_t computeStaticChunkSize(size_t totalWork,
   return roundUpDiv(workPerThread, chunksPerThread);
 }
 
+/*!
+ * \brief Copy data from one array-like object to another in parallel.
+ * \param[in] size - Number of elements.
+ * \param[in] src - Source array.
+ * \param[in] dst - Destination array.
+ */
+template<class T, class U>
+void parallelCopy(size_t size, const T* src, U* dst)
+{
+  SU2_OMP_FOR_STAT(4196)
+  for(size_t i=0; i<size; ++i) dst[i] = src[i];
+}
+
+/*!
+ * \brief Set the entries of an array-like object to a constant value in parallel.
+ * \param[in] size - Number of elements.
+ * \param[in] val - Value to set.
+ * \param[in] dst - Destination array.
+ */
+template<class T, class U>
+void parallelSet(size_t size, T val, U* dst)
+{
+  SU2_OMP_FOR_STAT(4196)
+  for(size_t i=0; i<size; ++i) dst[i] = val;
+}
+
+/*!
+ * \brief Atomically update a (shared) lhs value with a (local) rhs value.
+ * \note For types without atomic support (non-arithmetic) this is done via critical.
+ * \param[in] rhs - Local variable being added to the shared one.
+ * \param[in,out] lhs - Shared variable being updated.
+ */
+template<class T,
+         typename std::enable_if<!std::is_arithmetic<T>::value,bool>::type = 0>
+inline void atomicAdd(T rhs, T& lhs)
+{
+  SU2_OMP_CRITICAL
+  lhs += rhs;
+}
+template<class T,
+         typename std::enable_if<std::is_arithmetic<T>::value,bool>::type = 0>
+inline void atomicAdd(T rhs, T& lhs)
+{
+  SU2_OMP_ATOMIC
+  lhs += rhs;
+}
```

Comment:
Another small bit of probably unusual code for most.
This is SFINAE (substitution failure is not an error), the bottom function is active for arithmetic types (int, long, floats, doubles, etc.) for which atomic operations are defined by the OpenMP standard.
The top function will be used otherwise, using a critical region is slower but compatible with any type, which allows writing code that compiles with the AD types without #ifdefs in the code.
Forward mode AD + OpenMP is now enabled.